src/Tools.VisualStudio/InvokeProcessCommand.cs

```
@@ -0,0 +1,250 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Management.Automation;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.EntityFrameworkCore.Tools.VisualStudio
+{
+    /// <summary>
+    ///     <para>
+    ///         Invokes a new process and pipes output through to the PowerShell host.
+    ///         By default, this merges stdout and stderr into the 'Output' stream.
+    ///         (By comparison, Invoke-Expression and the call operator redirect stderr into the Error stream).
+    ///     </para>
+    ///     <para>
+    ///         When -RedirectByPrefix is specified, each line of stdout/stderr can be redirected to 'Debug', 'Verbose', etc. based on the prefix.
+    ///         <cref see="RedirectByPrefix" />.
+    ///         When -JsonOutput is specified, the cmdlet will only output JSON objects found in stdout/stderr that have been wrapped by '//BEGIN'
+    ///         and '//END' <cref see="JsonOutput" />
+    ///     </para>
+    /// </summary>
+    [Cmdlet("Invoke", "Process")]
+    [OutputType(typeof(string))]
+    public class InvokeProcessCommand : PSCmdlet
```

Here's what I've found so far:

- binary PS modules cannot be unloaded. 
  Implications:
  - `Uninstall-Package` tools package requires VS restart b/c the .dll is still loaded in the PMC domain
  - `Remove-Module` silently fails to remove binary modules. => Upgrading 1.1 to 1.2 (without a VS restart) will still have the old version of the binary module. We could mitigate by adding full version to the name of the name of the binary module.